operatorcert/entrypoints/create_container_image.py

```diff
@@ -0,0 +1,177 @@
+    )
+    parser.add_argument(
+        "--skopeo-result",
+        help="File with result of `skopeo inspect` running against image"
+        " represented by ContainerImage to be created",
+        required=True,
+    )
+    parser.add_argument(
+        "--image-size", help="sum of the size of image layers", required=True
+    )
+    parser.add_argument("--is-latest", help="Is given version latest?", required=True)
+    parser.add_argument("--verbose", action="store_true", help="Verbose output")
+    return parser
+
+
+def check_if_image_already_exists(args) -> bool:
+    check_url = urljoin(
+        args.pyxis_url,
+        f"v1/images?page_size=1&"
+        f"filter="
+        f"isv_pid=={args.isv_pid};"
+        f"docker_image_digest=={args.docker_image_digest};"
+        f"not(deleted==true)",
+    )
+
+    # Get the list of the ContainerImages with given parameters
+    rsp = pyxis.get(check_url)
+    rsp.raise_for_status()
+
+    query_results = rsp.json()["data"]
+
+    if len(query_results) == 0:
+        LOGGER.info(
+            "Image with given docker_image_digest and isv_pid doesn't exists yet"
+        )
+        return False
+
+    LOGGER.info(
+        "Image with given docker_image_digest and isv_pid already exists."
+        "Skipping the image creation."
+    )
+    return True
+
+
+def prepare_parsed_data(skopeo_result_file: str) -> Dict[str, Any]:
+    with open(skopeo_result_file) as json_file:
+        skopeo_result = json.load(json_file)
+
+        parsed_data = {
+            "docker_version": skopeo_result.get("DockerVersion", ""),
+            "layers": skopeo_result.get("Layers", []),
+            "architecture": skopeo_result.get("Architecture", ""),
+            "env_variables": skopeo_result.get("Env", []),
+        }
+
+        return parsed_data
+
+
+def create_container_image(args):
+    LOGGER.info("Creating new container image")
+
+    date_now = datetime.now().strftime("%Y-%m-%dT%H:%M:%S.%f+00:00")
+    parsed_data = prepare_parsed_data(args.skopeo_result)
+
+    upload_url = urljoin(args.pyxis_url, f"v1/images")
+    container_image_payload = {
+        "isv_pid": args.isv_pid,
+        "repositories": [
+            {
+                "published": True,
+                "registry": args.registry,
+                "repository": args.repository,
+                "push_date": date_now,
+                "tags": [
+                    {
+                        "added_date": date_now,
+                        # suffix -1 is added to indicate,
+                        # that it's non- floating tag
+                        "name": args.bundle_version + "-1",
+                    },
+                ],
+            }
+        ],
+        "certified": True,
+        "docker_image_digest": args.docker_image_digest,
+        "architecture": parsed_data["architecture"],
+        "parsed_data": parsed_data,
+        "sum_layer_size_bytes": int(args.image_size),
+    }
+
+    if args.is_latest == "true":
```

Comment:
Consider using the `action='store_true'` kwarg for `--is-latest` so that you can work with a boolean here.

Reply:
If it would be a `store_true` kwarg, then I don't know how could we comfortably optionally pass it from the Tekton task. Parameter `is_latest` is set on the level of the pipeline.

Reply:
Ah ok. Makes sense.